wezterm-ssh/src/config.rs

```diff
@@ -293,7 +295,14 @@ impl ParsedConfigFile {
                     let mut criteria = vec![];
                     let mut context = Context::FirstPass;
 
-                    let mut tokens = v.split_ascii_whitespace();
+                    let mut tokens = v.split_ascii_whitespace().map_while(|tok| {
```

Comment:
Hmm, I don't think this works as you intend, because I think this results in ignoring only the `# fun` token but not the `games` token from your expanded test case?

A more robust alternative would be to do something like:

```rust
let v, _commented = v.split_once('#').unwrap_or(v);
let mut tokens = v.split_ascii_whitespace();
```

Please expand the test case to explicitly verify the effective values for the `games` host so that we can prove/disprove this.